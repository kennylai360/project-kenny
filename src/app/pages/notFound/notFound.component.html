<div class="not-found">
  <h1>
    <i class="fa fa-exclamation-triangle" aria-hidden="true"></i>&nbsp;<span
      class="four-oh-four"
      >404</span
    >
    Not found&nbsp;<i class="fa fa-exclamation-triangle" aria-hidden="true"></i>
  </h1>
  <h4>The requested URL {{ requestedUrl }} does not exist!</h4>
  <span>That's an error.</span>
</div>
