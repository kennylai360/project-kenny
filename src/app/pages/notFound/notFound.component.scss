:host {
  .not-found {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    height: 80vh;
    text-align: center;
  }

  .four-oh-four {
    font-weight: 500;
    text-shadow: 0 0 4px #fff;
  }
}
