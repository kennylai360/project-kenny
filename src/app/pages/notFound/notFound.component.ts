import { CommonModule } from '@angular/common';
import { Component } from '@angular/core';
import { Title } from '@angular/platform-browser';
import { Router } from '@angular/router';
import { FontAwesomeModule } from '@fortawesome/angular-fontawesome';
import { faTriangleExclamation } from '@fortawesome/free-solid-svg-icons';

@Component({
    selector: 'app-not-found',
    templateUrl: './notFound.component.html',
    styleUrls: ['./notFound.component.scss'],
    imports: [CommonModule, FontAwesomeModule]
})
export class NotFoundComponent {
  public requestedUrl: string;

  protected icons = {
    notFound: faTriangleExclamation,
  };

  public constructor(private router: Router, private titleService: Title) {
    this.requestedUrl = this.router.url;
    this.titleService.setTitle('Error 404 - Page not found!');
  }
}
