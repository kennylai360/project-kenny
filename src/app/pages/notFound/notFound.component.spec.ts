import { TestBed } from '@angular/core/testing';
import { NotFoundComponent } from './notFound.component';

describe('NotFoundComponent', () => {
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [NotFoundComponent],
    }).compileComponents();
  });

  it('should create the app', () => {
    const fixture = TestBed.createComponent(NotFoundComponent);
    const app = fixture.componentInstance;
    expect(app).toBeTruthy();
  });

  it('should do a snapshot test', () => {
    const fixture = TestBed.createComponent(NotFoundComponent);
    expect(fixture.nativeElement).toMatchSnapshot();
  });
});
