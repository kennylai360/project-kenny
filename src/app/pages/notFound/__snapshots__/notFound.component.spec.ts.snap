// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundComponent should do a snapshot test 1`] = `
<div
  id="root1"
>
  <div
    class="not-found"
  >
    <h1>
      <fa-icon
        class="ng-fa-icon"
      />
       
      <span
        class="four-oh-four"
      >
        404
      </span>
       Not found 
      <fa-icon
        class="ng-fa-icon"
      />
    </h1>
    <h4>
      
    </h4>
    <span>
      That's an error.
    </span>
  </div>
</div>
`;
